docs/shared/packages/linter/dependency-checks.md

```diff
@@ -0,0 +1,71 @@
+# Dependency Checks rule
+
+The `@nx/dependency-checks` ESLint rule enables you to discover mismatches between dependencies specified in project's `package.json` and the dependencies that your project depends on.
+
+This rule will use the project graph to collect all the dependencies of your project, based on the input of your `build` target. It will filter out all the dependencies marked as `devDependencies` in your root `package.json` to ensures dependencies of your compilation pipelines (e.g. dependencies of `webpack.config` or `vite.config`) or test setups are not included in the expected list.
+
+We use versions of the installed packages to ensure the version specifier matches the version you are using while testing the functionality.
```

Comment:
I'm confused by this sentence.  I'm not quite sure what you're trying to communicate, but this could be an implementation detail that could just be removed.  Or if its important, I need to understand what "while testing the functionality" means.

Reply:
So reason behind this is that specified range should include the installed version, as that is the version for which you can "guarantee" that things work. If you specify range outside of that version, that means you couldn't have possibly tested it with.

Reply:
Ok, that makes sense to include in the docs.  In that case, you'll need a paragraph explaining it the way you just did.